aspnet-core/src/TalentV2.Core/BackgroundWorker/CVTransformHelper/CVAutomationFirebase.cs

```diff
@@ -0,0 +1,358 @@
+            };
+            return true;
+        }
+
+
+
+        private async Task SaveCVToAWS(string fileURL)
+        {
+            using (var client = _httpClientFactory.CreateClient())
+            {
+                var fileName = Path.GetFileName(new Uri(fileURL).AbsolutePath);
+                var fileNameDecoded = WebUtility.UrlDecode(fileName);
+                var fileBytes = await client.GetByteArrayAsync(fileURL);
+                var streamConverted = new MemoryStream(fileBytes);
+                var formFile = CreateFormFile(fileBytes, fileNameDecoded, streamConverted);
+                var paths = await _filePath.GetPath(FOLDER_SERVICE, PathFolder.FOLDER_CV, _abpSession.TenantId);
+                await _fileServiceAWS.UploadFileAsync(paths, formFile);
+            };
+
+        }
+
+        private static IFormFile CreateFormFile(byte[] fileBytes, string fileName, MemoryStream stream)
+        {
+            return new FormFile(stream, 0, fileBytes.Length, "file", fileName)
+            {
+                Headers = new HeaderDictionary(),
+                ContentType = "application/octet-stream" // 
+            };
+
+
+
+
+        }
+        private async Task<Position> GetOrCreatePositionAsync(string positionName)
+        {
+            var position = await _repositoryPosition.FirstOrDefaultAsync(p =>
+                p.Name.Equals(positionName));
+            if (position == null)
+            {
+
+                using (var unitOfWorkPosition = _unitOfWorkManager.Begin(System.Transactions.TransactionScopeOption.RequiresNew))
+                {
+                    position = new Position { Name = positionName };
+                    await _repositoryPosition.InsertAsync(position);
+                    await unitOfWorkPosition.CompleteAsync();
+                };
+            }
+            return position;
+        }
+        private async Task<SubPosition> GetOrCreateSubPositionAsync(string subPositionName)
+        {
+            var defaultSubPosition = await _repositorySubPosition.FirstOrDefaultAsync(p => p.Name.Equals(subPositionName));
+            if (defaultSubPosition == null)
+            {
+                var position = await GetOrCreatePositionAsync("NotCreatedPosition");
+
+                using (var unitOfWorkSubPosition = _unitOfWorkManager.Begin(System.Transactions.TransactionScopeOption.RequiresNew))
+                {
+                    defaultSubPosition = new SubPosition
+                    {
+                        Name = "NeedToUpdate",
+                        PositionId = position.Id,
+
+                    };
+                    await _repositorySubPosition.InsertAsync(defaultSubPosition);
+                    await unitOfWorkSubPosition.CompleteAsync();
+                };
+            }
+
+            return defaultSubPosition;
+        }
+        private async Task<CVScanResulFromFireBase> IsAcceptedFileAsync(HttpClient client, string fileUrl, string firebaseId)
```

Comment:
đổi tên hàm cho có ý nghĩa hơn
`private async Task<CVScanResulFromFireBase> IsAcceptedFileAsync(HttpClient client, string fileUrl, string firebaseId)`

1. Nếu không return boolean tốt nhất không nên đặt là Is...
2. Đã là một hàm để check đúng lại mà lại insert data vào database, không sai logic máy, nhưng sai logic người